src/huggingface_hub/file_download.py

```diff
@@ -505,7 +505,7 @@ def http_get(
         total=total,
         initial=resume_size,
         desc=displayed_filename,
-        disable=bool(logger.getEffectiveLevel() == logging.NOTSET),
+        disable=True if (logger.getEffectiveLevel() == logging.NOTSET) else None,
```

Comment:
```suggestion
        disable=True if (logger.getEffectiveLevel() == logging.NOTSET) else None,
        # ^ set `disable=None` rather than `disable=False` by default to disable progress bar when no TTY attached
        # see https://github.com/huggingface/huggingface_hub/pull/2000
```

^ could you add a comment that `disable=None` has an implicit meaning in this context?

Reply:
Sure

Reply:
Does the project have a limit on line length?

Reply:
I pushed the fix for `lfs.py`. My question is whether or not it would be preferable to make this change in `utils/tqdm.py` instead. The question is whether you'd rather force `True or None` to be the behavior. This has the benefit of ensuring this bug doesn't creep in somewhere else, but the drawback of changing the interface of `tqdm` throughout `huggingface_hub`.

```diff
diff --git a/src/huggingface_hub/utils/tqdm.py b/src/huggingface_hub/utils/tqdm.py
index cae9adb..c3dddad 100644
--- a/src/huggingface_hub/utils/tqdm.py
+++ b/src/huggingface_hub/utils/tqdm.py
@@ -129,6 +129,10 @@ class tqdm(old_tqdm):
     def __init__(self, *args, **kwargs):
         if are_progress_bars_disabled():
             kwargs["disable"] = True
+
+        if kwargs["disable"] is False:
+            kwargs["disable"] = None
+
         super().__init__(*args, **kwargs)
 
     def __delattr__(self, attr: str) -> None:
```

Reply:
I would prefer not to do that, just to keep the `disable=False` behavior possible (can't find a useful use case but you never know). Thanks for suggesting though, we might reevaluate in the future.